src/cs50/cs50.py

```diff
@@ -76,6 +77,9 @@ def get_char(prompt=None):
     if text is not a single char, user is prompted to retry. If line can't
     be read, return None.
     """
+    caller = getframeinfo(stack()[1][0])
+    cprint(warnings.formatwarning("get_char will soon be deprecated, use get_string instead",
```

Comment:
I think _normally_ the right thing to do here would be to call `warning.simplefilter("once", PendingDeprecationWarning)` at the top of the file, and then call `warning.warn("get_char will soon ...", PendingDeprecationWarning, stacklevel=2)` from inside  `get_char`. This way the warning would only be printed once instead of every time `get_char` is called (e.g. if it were called in a loop) and you wouldn't have to do the stack frame parsing yourself, but then you wouldn't get the yellow text, so this is probably the best option